source/client/benchmark_client_impl.h

```diff
@@ -66,6 +67,48 @@ class Http1PoolImpl : public Envoy::Http::Http1::ProdConnPoolImpl {
   bool prefetch_connections_{};
 };
 
+// Vanilla Envoy's HTTP/2 pool is single connection only (or actually sometimes dual in connection
+// drainage scenarios). Http2PoolImpl is an experimental pool, which uses multiple vanilla Envoy
+// HTTP/2 pools under the hood. Using multiple connections is useful when testing backends that need
+// multiple connections to distribute load internally. Combining multiple connections with
+// --max-requests-per-connection may help as well, as doing periodically initiating new connections
+// may help the benchmark target by giving it an opportunity to rebalance.
+class Http2PoolImpl : public Envoy::Http::ConnectionPool::Instance,
+                      public Envoy::Http::ConnPoolImplBase {
+public:
+  Http2PoolImpl(
+      Envoy::Event::Dispatcher& dispatcher, Envoy::Upstream::HostConstSharedPtr host,
+      Envoy::Upstream::ResourcePriority priority,
+      const Envoy::Network::ConnectionSocket::OptionsSharedPtr& options,                // NOLINT
```

Comment:
Why do we need the "NOLINT"s here?

Reply:
clang-tidy doesn't like this [1]. But as the argument list is implied by a base class that originates from Envoy, it is hard to change this according to what it wants.

[1] https://app.circleci.com/jobs/github/envoyproxy/nighthawk/9194

```
clang-tidy -extra-arg-before=-xc++ -p=/root/project -quiet /build/tmp/_bazel_bazel/f85b6fb5740e6e8c7efea142eec4b6e8/execroot/nighthawk/source/client/benchmark_client_impl.cc
/build/tmp/_bazel_bazel/f85b6fb5740e6e8c7efea142eec4b6e8/execroot/nighthawk/source/client/benchmark_client_impl.cc:50:5: error: pass by value and use std::move [modernize-pass-by-value,-warnings-as-errors]
    const Envoy::Network::ConnectionSocket::OptionsSharedPtr& options,
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Envoy::Network::ConnectionSocket::OptionsSharedPtr 
/build/tmp/_bazel_bazel/f85b6fb5740e6e8c7efea142eec4b6e8/execroot/nighthawk/source/client/benchmark_client_impl.cc:51:5: error: pass by value and use std::move [modernize-pass-by-value,-warnings-as-errors]
    const Envoy::Network::TransportSocketOptionsSharedPtr& transport_socket_options)
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Envoy::Network::TransportSocketOptionsSharedPtr 
14185 warnings generated.
```